CDTDatastore/HTTP/CDTRequestLimitInterceptor.m

```diff
@@ -50,16 +49,32 @@ - (instancetype)initWithSleep:(NSTimeInterval)sleep
  */
 - (CDTHTTPInterceptorContext *)interceptResponseInContext:(CDTHTTPInterceptorContext *)context
 {
-    if (context.response.statusCode == 429 && self.maxRequests > 0) {
+    if (context.response.statusCode == 429) {
+
+        // if we are the first invocation in this pipeline, set some state
+        if (!context.state[@"sleep"]) {
```

Comment:
Agreed, or instead of exposing the dictionary I expose `getValue` and `setValue` which operate on the underlying dictionary and use some sneaky obj-c magic (if possible) to automatically prepend with the interceptor class name